module/actor.js

```diff
@@ -158,6 +164,42 @@ class DCCActor extends Actor {
 
     return `<a class="${critFailClass}inline-roll inline-result" data-roll="${rollData}" title="${formula}"><i class="fas fa-dice-d20"></i> ${roll.total}</a>`
   }
+
+  /**
+   * Apply damage to this actor
+   * @param {Number} weaponId     Damage amount to apply
+   * @param {Number} multiplier   Damage multiplier
+   */
+  applyDamage (damageAmount, multiplier) {
+    const speaker = { alias: this.name, _id: this._id }
+
+    // Calculate damage amount and current hit points
+    const amount = damageAmount * multiplier
+    const hp = this.data.data.attributes.hp.value
+
+    // Deduct damage from hit points
+    let newHp = hp - amount
+
+    // Clamp at max HP
```

Comment:
I'm wondering if perhaps we only want to clamp going up, and not going down? e.g. in the case where someone has temp hp, or something like that?

Reply:
Good call.

Took me a second, but you're right, this needs to be a bit smarter to avoid corrupting the players hit points if they were above max for some reason.

Reply:
I think this now handles temporary hp correctly.

I suspect the messaging and maybe the assumptions about negative hp will need a revisit at some point, now it only reports if the hit point total changed as reporting damage/healing for zero felt wrong.

Reply:
Temp HP looks good now! Agreed...long term would be good to indicate to the GM and token owner that their new hp total.